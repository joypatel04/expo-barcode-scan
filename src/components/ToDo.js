'use strict';

import React, { PropTypes, Component} from 'react';
import { StyleSheet, Text, View, TouchableOpacity, ScrollView, TextInput } from 'react-native';
import { FontAwesome } from '@expo/vector-icons'
import { ActionCreators } from './../actions';
import Store from './../store/configureStore';

export default class ToDo extends Component {

    constructor(props) {
        super(props);
        this.state = {
            toDoText: '',
            toDos : this.props.toDos,
            viewWidth: 0
        } 
    }

    render() {
        return (
            <ScrollView contentContainerStyle={[styles.container, {width: this.props.width}]} scrollEnabled={false}>
                <View style={[styles.header]}>
                    <Text style={styles.headerTitle}>To-Do Example</Text>
                </View>
                <View style={[styles.textInputView]}>
                    <TextInput
                        ref="toDoInput"
                        selectionColor="rgba(0,80,255,0.90)"
                        style={[styles.textInput, {width: this.props.width - 20}]}
                        maxLength={30}
                        value={this.state.toDoText}
                        onChangeText={(toDoText) => this.setState({toDoText: toDoText})}
                        placeholder="Add To-Do"
                        keyboardAppearance="dark"
                        autoCapitalize="words"
                        autoCorrect={false}
                        returnKeyType="done"
                    />
                </View>
            </ScrollView>    
        );
    }
}

ToDo.propTypes = {
    toDos: PropTypes.object.isRequired
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: window.width,
    backgroundColor: 'transparent',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
  header: {
    padding: 10
  },
  headerTitle: {
    fontFamily: 'HelveticaNeue',
    fontSize: 16,
    fontWeight: '700',
    color: '#455A64',
    letterSpacing: 0
  },
  textInputView: {
    marginRight: 2,   
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 10,
    borderWidth: 0.5,
    borderColor: '#CFD8DC'
  },
  textInput: {
    textAlign: 'center',
    height: 40,
    fontSize: 16,
    color: '#455A64',
    fontFamily: 'HelveticaNeue',
    fontWeight: '500',
  }
});