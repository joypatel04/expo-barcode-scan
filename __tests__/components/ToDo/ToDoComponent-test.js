import 'react-native';
import React from 'react';
import ToDo from './../../../src/components/ToDo';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('Render CounterExample Component Correctly', () => {
    const tree = renderer.create(
        <ToDo toDos={{todos: []}} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
})