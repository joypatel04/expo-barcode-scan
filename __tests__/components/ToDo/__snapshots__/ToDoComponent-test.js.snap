// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render CounterExample Component Correctly 1`] = `
<ScrollView
  contentContainerStyle={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "justifyContent": "flex-start",
        "width": undefined,
      },
      Object {
        "width": undefined,
      },
    ]
  }
  scrollEnabled={false}
>
  <View
    style={
      Array [
        Object {
          "padding": 10,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#455A64",
          "fontFamily": "HelveticaNeue",
          "fontSize": 16,
          "fontWeight": "700",
          "letterSpacing": 0,
        }
      }
    >
      To-Do Example
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#CFD8DC",
          "borderRadius": 10,
          "borderWidth": 0.5,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginRight": 2,
        },
      ]
    }
  >
    <TextInput
      autoCapitalize="words"
      autoCorrect={false}
      keyboardAppearance="dark"
      maxLength={30}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Add To-Do"
      returnKeyType="done"
      selectionColor="rgba(0,80,255,0.90)"
      style={
        Array [
          Object {
            "color": "#455A64",
            "fontFamily": "HelveticaNeue",
            "fontSize": 16,
            "fontWeight": "500",
            "height": 40,
            "textAlign": "center",
          },
          Object {
            "width": NaN,
          },
        ]
      }
      value=""
    />
  </View>
  <ScrollView
    contentContainerStyle={
      Array [
        Object {
          "marginRight": 2,
          "width": NaN,
        },
      ]
    }
    dataSource={
      ListViewDataSource {
        "items": 0,
      }
    }
    enableEmptySections={true}
    initialListSize={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={1000}
    onKeyboardDidHide={undefined}
    onKeyboardDidShow={undefined}
    onKeyboardWillHide={undefined}
    onKeyboardWillShow={undefined}
    onLayout={[Function]}
    onScroll={[Function]}
    pageSize={1}
    removeClippedSubviews={true}
    renderRow={[Function]}
    scrollEventThrottle={50}
    scrollRenderAheadDistance={1000}
    stickyHeaderIndices={Array []}
    stickySectionHeadersEnabled={true}
  />
</ScrollView>
`;
